commons/src/main/java/org/infinispan/commons/util/Util.java

```diff
@@ -1027,6 +1028,14 @@ else if (ch == 'f')
       return out.toString();
    }
 
+   public static <T> Supplier<T> getInstanceSupplier(Class<T> klass) {
```

Comment:
Why not `Function` instead of `Supplier`? By using a Supplier, you're capturing a variable outside the lambda (klass). Could you have a Function that takes `Class<T>` and returns `T`? Then the function would be pure.

Reply:
Because in the context of the server LocalConfigurationStorage, I need to be able to set a bunch of context.